### PR TITLE
added typing IntervalClosedType where needed

### LINKED ISSUE #48140: TYP: Make typing consistent for closed argument in Interval associated methods/classes
Initially, this was fixed in https://github.com/pandas-dev/pandas/pull/47646 but had to be reverted to rename inclusive back to closed

### PR DESCRIPTION
**Pull request by gmollard**
- [x] closes #48140  (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

@phofl

### CODE REVIEW

File: pandas/core/dtypes/dtypes.py [hunk 1]
```diff
@@ -1099,7 +1100,7 @@ class IntervalDtype(PandasExtensionDtype):
 
     _cache_dtypes: dict[str_type, PandasExtensionDtype] = {}
 
-    def __new__(cls, subtype=None, closed: str_type | None = None):
+    def __new__(cls, subtype=None, closed=None):
```
Comment (phofl) [inline pandas/core/dtypes/dtypes.py:1103]: Could you add it here too?

Comment (phofl) [reply pandas/core/dtypes/dtypes.py:1103]: Adding ``# type: ignore[assignment] `` in 1141 would work here. This isn't ideal, but it's better if the function signature is typed more strictly

Comment (gmollard) [reply pandas/core/dtypes/dtypes.py]: I added what you asked but I have a check that fails, I don't know why..
Do you have an idea of what's the reason?

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx @gmollard

**Pull request merged**